vpr/src/analysis/timing_reports.cpp

```diff
@@ -61,3 +183,76 @@ void generate_hold_timing_stats(const std::string& prefix,
 
     timing_reporter.report_unconstrained_hold(prefix + "report_unconstrained_timing.hold.rpt", *timing_info.hold_analyzer());
 }
+
+void generate_net_timing_report(const std::string& prefix,
+                                const SetupHoldTimingInfo& timing_info,
+                                const AnalysisDelayCalculator& delay_calc) {
+    /* Create a report file for net timing information */
+    std::ofstream os(prefix + "report_net_timing.rpt");
+    const auto& atom_netlist = g_vpr_ctx.atom().netlist();
+    const auto& atom_lookup = g_vpr_ctx.atom().lookup();
+
+    const auto& timing_ctx = g_vpr_ctx.timing();
+    const auto& timing_graph = timing_ctx.graph;
+
+    os << "# This file is generated by VTR" << std::endl;
+    os << "# Version: " << vtr::VERSION << std::endl;
+    os << "# Revision: " << vtr::VCS_REVISION << std::endl;
+    os << "# For each net, the timing information is reported in the following format:" << std::endl;
+    os << "# netname : Fanout : "
+       << "(bounding_box_xmin,bounding_box_ymin,bounding_box_layermin),(bounding_box_xmax,bounding_box_ymax,bounding_box_layermax) : "
+       << "source_instance <slack_on source pin> : "
+       << "<load pin name1> <slack on load pin name1> <net delay for this net> : "
+       << "<load pin name2> <slack on load pin name2> <net delay for this net> : ..."
+       << std::endl;
+
+    os << std::endl;
+
+    for (const auto& net : atom_netlist.nets()) {
+        /* Skip constant nets */
+        if (atom_netlist.net_is_constant(net)) {
+            continue;
```

Comment:
Why not just print them out?  If they were routed they'll be printed and if not they'll be skipped.